pkg/skaffold/schema/latest/config.go

```diff
@@ -75,13 +75,28 @@ type Pipeline struct {
 	PortForward []*PortForwardResource `yaml:"portForward,omitempty"`
 }
 
+// GitInfo contains information on the origin of skaffold configurations cloned from a git repository.
+type GitInfo struct {
+	// Repo is the git repository the package should be cloned from.  e.g. `https://github.com/GoogleContainerTools/skaffold.git`.
+	Repo string `yaml:"repo" yamltags:"required"`
+
+	// Path is the relative path from the repo root to the skaffold configuration file. eg. `getting-started/skaffold.yaml`.
+	Path string `yaml:"path,omitempty"`
+
+	// Ref is the git ref the package should be cloned from. eg. `master` or `main`.
+	Ref string `yaml:"ref,omitempty"`
+}
+
 // ConfigDependency describes a dependency on another skaffold configuration.
 type ConfigDependency struct {
 	// Names describes the names of the required configs.
 	Names []string `yaml:"configs,omitempty"`
```

Comment:
Can we improve this:
```suggestion
	// Names includes specific named configs within the file path. If empty, then all configs in the file are included.
	Names []string `yaml:"configs,omitempty"`
```

Reply:
done